http/sys_seal.go

```diff
@@ -174,25 +174,30 @@ func handleSysSealBackendStatus(core *vault.Core) http.Handler {
 	})
 }
 
-func handleSysSealStatusRaw(core *vault.Core, w http.ResponseWriter, opt ...ListenerConfigOption) {
-	ctx := context.Background()
-	status, err := core.GetSealStatus(ctx, true)
-	if err != nil {
-		respondError(w, http.StatusInternalServerError, err)
-		return
+func handleSysSealStatusRaw(core *vault.Core, w http.ResponseWriter, r *http.Request, opt ...ListenerConfigOption) {
+	ctx := r.Context()
+
+	var tokenPresent bool
+	token := r.Header.Get(consts.AuthHeaderName)
+	if token != "" {
+		// We don't care about the error, we just want to know if the token exists
+		lock := core.HALock()
+		lock.Lock()
+		tokenEntry, err := core.LookupToken(ctx, token)
+		lock.Unlock()
+		tokenPresent = err == nil && tokenEntry != nil
 	}
 
-	opts, err := getOpts(opt...)
-
-	if opts.withRedactVersion {
-		status.Version = opts.withRedactionValue
-		status.BuildDate = opts.withRedactionValue
+	// If there are is no valid token then we will redact the specified values
+	if tokenPresent {
```

Comment:
Similar issue here. You need to determine if either `opts.withRedactVersion` or `opts.withRedactClusterName` is `true`. Then if there is no token present (i.e. `!tokenPresent) and at least one of those 2 redaction settings are `true` (i.e. `opts.withRedactVersion || opts.withRedactClusterName`), a wrapping context needs to be created to pass down the redaction settings to `core.GetSealStatus`. There's never a need to call `logical.CreateContextRedactionSettings(ctx, false, false, false)` since the function behaves the same as if no wrapping context was sent.

Reply:
Like the comment above, I retract my comment since I fail to appreciate that the redactionSettingsWrapping apply to the entire mux not a subset of them.